app/javascript/react/screens/App/Plan/components/PlanRequestDetailList/PlanRequestDetailList.js

```diff
@@ -270,6 +276,74 @@ class PlanRequestDetailList extends React.Component {
     }
   };
 
+  getCancelSelectionState = () => {
+    const { selectedTasksForCancel, markedForCancellation } = this.props;
+    const filteredTasks = this.filterPlanRequestTasks();
+    const incompleteTasks = filteredTasks.filter(
+      task => !task.completed && !task.cancel_requested && !markedForCancellation.find(t => t.id === task.id)
+    );
+    return {
+      filteredTasks,
+      incompleteTasks,
+      allSelected: selectedTasksForCancel.length === incompleteTasks.length,
```

Comment:
Ah, @AparnaKarve I didn't notice you based your work on a slightly older version of `getCancelSelectionState`.  I pushed a newer one on Wednesday that fixed the bug @michaelkro was just reporting (select all checkbox is checked when there are no selectable tasks).

We just need to pull `noneSelected` up into a variable, and then do `allSelected: !noneSelected && ...`